app/frontend/src/service/api.ts

```diff
@@ -5,7 +5,10 @@ import { ConversationsPromiseClient } from "../pb/conversations_grpc_web_pb";
 import { RequestsPromiseClient } from "../pb/requests_grpc_web_pb";
 import { SSOPromiseClient } from "../pb/sso_grpc_web_pb";
 
-import { store } from "../store";
+let getToken = () => "";
+export function setGetToken(newGetToken: () => string) {
+  getToken = newGetToken;
+}
```

Comment:
Looks like it's introduced by a potentially unnecessary import that I don't really follow the need for in `index.tsx`. If we are using a global state management tool like Redux, we really should stick to it as the single source of truth for global states rather than trying to synchronise it in random module states, otherwise this becomes hard to reason about as the app grows.

Reply:
Those circular imports were coming from the stories in story book, when trying to mock service methods.

Reply:
btw: We are not using a single source of truth for global states as there is also some global state in the url. We use useParams to retrieve that state.

Reply:
That's a good point, I'll backtrack a bit and should rephrase the single source of truth for a particular piece of state, which the old code broke by having it both in the Redux store and in the module and it's the main problem I had with it. Having multiple state sources isn't a problem (in fact, it's recommended by React like local component state) as long we rely on those to be the source of truth rather than sporadically syncing them all over the place.

Reply:
But didn't my solution of setting a getter function keep a single source of truth? Anyway: I removed it. So I think you could approve the code.